lib/cartopy/img_transform.py

```diff
@@ -354,25 +334,15 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
     x_extent = np.abs(target_proj.x_limits[1] - target_proj.x_limits[0])
     y_extent = np.abs(target_proj.y_limits[1] - target_proj.y_limits[0])
 
-    non_self_inverse_points = (np.abs(target_x_points - back_to_target_x) /
-                               x_extent) > FRACTIONAL_OFFSET_THRESHOLD
+    non_self_inverse_points = (((np.abs(target_x_points - back_to_target_x) /
+                                x_extent) > FRACTIONAL_OFFSET_THRESHOLD) |
+                               ((np.abs(target_y_points - back_to_target_y) /
```

Comment:
Was this necessary, or is it in the interest of symmetry? If the latter, I'd vote to remove it, as it has a performance implication.

Reply:
It already happens below; this is consolidating it into one array and one check/application of the mask.